src/common/ownsql.h

```diff
@@ -136,10 +136,21 @@ class OCSYNC_EXPORT SqlQuery
     template <class T>
     void bindValue(int pos, const T &value)
     {
-        QString s;
-        QDebug(&s).noquote().nospace() << '\'' << value << '\'';
-        _boundQuery.replace(QStringLiteral("?%1").arg(QString::number(pos)), s);
-        bindValueConvert(pos, value);
+        const auto converted = convertValue(value);
+        if (lcSql().isDebugEnabled()) {
+            QString s;
+            {
+                auto stream = QDebug(&s).noquote().nospace() << '\'' << value;
+                if (typeid(converted) != typeid(value)) {
+                    stream << " [" << converted << ']';
+                }
+                stream << '\'';
+            }
+            const QString key = QStringLiteral("?%1").arg(QString::number(pos));
```

Comment:
How to handle the case that a value contained a ?5

Reply:
Not sure there's a good way. The approach of caching the rendered statement in a string is obviously flawed. There are multiple cases in which this is unreliable, e.g., when the first bound value contains another `?2`. Maybe the caching should be removed altogether? You could store the bound values as strings in a map, then write log messages like `<statement>: bound value <value> to position <position>, bound values: <map>`.